lib/ExpensiMark.js

```diff
@@ -346,6 +348,39 @@ export default class ExpensiMark {
         return replacedText;
     }
 
+    /**
+     * Map the block to newline
+     *
+     * @param {String} htmlString
+     *
+     * @returns {String}
+     */
+    mapBlock(htmlString) {
+        const splitText = htmlString.split('[block]');
+        let joinedText = '';
+
+        // delete empty spaces at the end
+        while (splitText.length) {
+            if (splitText[splitText.length - 1].trim().length > 0 || splitText[splitText.length - 1].match(/\n/)) {
+                break;
+            }
+            splitText.pop();
+        }
+
+        splitText.forEach((text, index) => {
+            if (text.trim().length === 0 && !text.match(/\n/)) {
+                return;
+            }
+            if (text.match(/[\n|>][>]?[\s]?$/) || index === splitText.length - 1) {
```

Comment:
ok, we have some cases we do not add a new line. 
1- if we have a newline at the end. 
2- if we are in the last div. 
3- if we have a quote we now map "> " to newline since it's also inside block

Reply:
Thanks! Please add a comment for the same.